src/Autofac/ResolutionExtensions.cs

```diff
@@ -194,6 +194,10 @@ public static bool IsRegistered(this IComponentContext context, Type serviceType
     /// <returns>True if the service is registered.</returns>
     public static bool IsRegisteredService(this IComponentContext context, Service service)
     {
+#if NET7_0_OR_GREATER
```

Comment:
I don't like adding a load of additional `#if` blocks like this just to optimise the null check. Can you provide a benchmark with evidence of the gain?  It would have to be a pretty big improvement to justify I think.

Reply:
> Can you provide a benchmark with evidence of the gain

It will be difficult to model the impact of CPU I-cache pollution on a real application.
Typical benchmarks contain little code and CPU cache is big enough.
And big-code benchmark would have many other, may be volatile, factors 

>I don't like adding a load of additional #if

There is other way:  to write custom `ThrowHelper` with the same functionality as `ArgumentNullException.ThrowIfNull()`

Reply:
We've definitely avoided adding the ThrowIfNull thing in the past simply because the maintenance overhead has outweighed the micro-perf you might get from that helper. I don't think they do this in the core framework libraries, either.

Reply:
> I don't think they do this in the core framework libraries

They do this from beginning of the "Core fork". e.g.
https://github.com/dotnet/runtime/blame/a0e9764708d5781168c22d86ccb7437f1e54a915/src/coreclr/src/mscorlib/src/System/Collections/Generic/Dictionary.cs#L106